### PR TITLE
161967139 Add tags to Articles

### PR DESCRIPTION
**Pull request by davishooly**
**What does this PR do?**

**Description of Task to be completed?**

- Add tags to articles.
- Updates tags for an article.

**How should this be manually tested?**

- Clone the repo and cd into the project folder
- Create a virtual environment by running $ python3 -m venv <name_of_environment>
- Activate the virtual environment by running source <name_of_environment>/bin/activate
- Run $ pip install -r requirements.txt
- Then add the following environment variables to your computer.

```
$ export DB_NAME='choose_a_dbname'
$ export DB_USER='db_username'
$ export DB_PASSWORD='db_role_password'
$ export DB_HOST='localhost or preffered host'
$ export DB_PORT='server_port_number'
$ export APP_DEBUG='True'
$ export SOCIAL_AUTH_FACEBOOK_KEY='<facebook_app_id>'
$ export SOCIAL_AUTH_FACEBOOK_SECRET='<facebook_app_secret>'
$ export SOCIAL_AUTH_TWITTER_KEY='<twitter_api_key>'
$ export SOCIAL_AUTH_TWITTER_SECRET='<twitter_api_secret>'
$ export SOCIAL_AUTH_GOOGLE_OAUTH2_KEY='<google_client_id>'
$ export SOCIAL_AUTH_GOOGLE_OAUTH2_SECRET='<google_client_secret>'
```

- Change the current branch to ft-taglist-161967139
- Run `python3 manage.py makemigrations`
- Run `python3 manage.py migrate`
- Run `python3 manage.py test`
- Run `python3 manage.py runserver`
- Test `http://localhost:8000/api/article/<slug>` with postman

**What are the relevant pivotal tracker stories?**
[#161967139](https://www.pivotaltracker.com/story/show/161967139)

**Screenshot
screen shot 2018-12-19 at 12 47 54

screen shot 2018-12-18 at 17 36 12
**

**

### CODE REVIEW

**Review by denomash — CHANGES_REQUESTED** [review]
@davishooly I noticed that while updating taglist, the validations are not working correctly. You could probably add validations on that since your post validations work fine.

**Comment by davishooly** [comment]
@denomash thanks for catching that up, I successfully made the changes

**Comment by denomash** [comment]
👍

**Review by denomash — APPROVED** [review]

File: authors/apps/articles/tests/test_get_article.py [hunk 1]
```diff
@@ -20,3 +20,7 @@ def test_get_all_articles(self):
         )
         self.assertEqual(response.status_code, status.HTTP_200_OK)
 
+
+
+
```
Comment (Bryan-Cee) [inline authors/apps/articles/tests/test_get_article.py]: Delete these extra lines

**Review by Bryan-Cee — CHANGES_REQUESTED** [review]

File: authors/apps/articles/tests/test_get_article.py (hunk 1 above)
Comment (davishooly) [reply authors/apps/articles/tests/test_get_article.py]: i have made the changes @Bryan-Cee

**Review by Bryan-Cee — APPROVED** [review]

File: authors/apps/articles/serializers.py [hunk 2]
```diff
@@ -28,10 +52,46 @@ def validate_user_permissions(self, request, data):
                 status=status.HTTP_403_FORBIDDEN
             )
 
+    def validate_tag_list(self, validated_data):
```
Comment (Darius-Ndubi) [inline authors/apps/articles/serializers.py:60]: Good work validating that the input is a List, I would request you kindly validate that the list items are strings

**Review by Darius-Ndubi — CHANGES_REQUESTED** [review]

File: authors/apps/articles/serializers.py (hunk 2 above)
Comment (davishooly) [reply authors/apps/articles/serializers.py:60]: @Darius-Ndubi I have made the changes you requested

**Review by Darius-Ndubi — APPROVED** [review]

**Review by mutoromike — APPROVED** [review]

**Pull request merged**